src/login.js

```diff
@@ -0,0 +1,65 @@
+import React, { useState} from "react";
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+import {createPatient} from './firebase'
+import { useHistory } from 'react-router-dom';
+import { jsPsych } from 'jspsych-react'
+
+ function Login() {
+    const dateTimestamp = Date.now()
+    const curDate = new Date(dateTimestamp)
+    const [patientId, setPatient] = useState("");
+    const [studyId, setStudy] = useState("")
+    const [startDate] = useState(curDate.toString())
+    const [timestamp] = useState(dateTimestamp.toString())
+    const history = useHistory();
+
+    function validateForm() {
+        return patientId.length > 0 && studyId.length > 0;
+      }
+
+    function handleSubmit(e) {
+        e.preventDefault();
+        const logPatient = async() =>{
+            const loggedIn = await createPatient(patientId, studyId, startDate, timestamp)
+            
+            if(loggedIn){
+                jsPsych.data.addProperties({patient_id: patientId, study_id: studyId, timestamp: timestamp, start_date: startDate})
+                history.push('/experiment')
```

Comment:
If made a component, this won't be necessary

Reply:
I'm confused - isn't it already a component? Instead this would need to pass the action to set logged in on the parent component, no?

Reply:
History is part of the `react-router` library. I think @suivo2001 is trying to preserve the browser back button behavior here. I think that's a good reason for a router, but in this case I still don't think it matters that much

Reply:
interesting - I think that's actually a really good reason to not use `router` in this case - participants shouldn't be able to go back.  We'd also need to protect `/experiment` so that one can't navigate to there without logging in first.

Reply:
... I meant if we render `<ExpStart>` directly component instead of routing to `/experiment`... I think we are all thinking of the same and I thing you make a good point about not being able to go back in the history nor navigate directly to the `/experiment` route